docs/csharp/roslyn-sdk/tutorials/how-to-write-csharp-analyzer-code-fix.md

```diff
@@ -0,0 +1,615 @@
+
+Throughout this tutorial, you'll use a combination of running your analyzer in a second copy of Visual Studio, and adding new unit tests that validate your analyzer and code fix. This combination provides a way to explore your analyzer's capabilities as you build it, and create quality tests to prevent regressions and validate actions quickly. This practice is often a good way to proceed with your analyzers: experiment in Visual Studio, and then write tests as you learn more about the code to spot and change.
+
+## Create the analyzer
+
+The template shows the basic features that are part of any analyzer:
+
+1. Register actions. The actions represent code changes that should trigger your analyzer to examine code for violations. When Visual Studio detects code edits that match a registered action, it calls your analyzer's registered method.
+1. Create diagnostics. When your analyzer detects a violation, it creates a diagnostic object that Visual Studio uses to notify the user of the violation.
+
+You register actions in your override of <xref:Microsoft.CodeAnalysis.Diagnostics.DiagnosticAnalyzer.Initialize(Microsoft.CodeAnalysis.Diagnostics.AnalysisContext)?displayProperty=nameWithType> method. The actions your register create the diagnostics. In this tutorial, you'll visit **syntax nodes** looking for local declarations, and see which of those have constant values.
+
+### Create the analyzer structure
+
+Open **MakeConstAnalyzer.cs** in Visual Studio. Change the registered action from one that acts on symbols to one that acts on syntax. In the `MakeConstAnalyzerAnalyzer.Initializer` method, find the line that registers the action on symbols:
+
+```csharp
+context.RegisterSymbolAction(AnalyzeSymbol, SymbolKind.NamedType);
+```
+
+Replace it with the following line:
+
+[!code-csharp[Register the node action](~/samples/csharp/roslyn-sdk/Tutorials/MakeConst/MakeConst/MakeConst/MakeConstAnalyzer.cs#RegisterNodeAction "Register a node action")]
+
+After that change, you can delete the `AnalyzeSymbol` method. This analyzer examines <xref:Microsoft.CodeAnalysis.CSharp.SyntaxKind.LocalDeclarationStatement?displayProperty=nameWithType>, not <xref:Microsoft.CodeAnalysis.SymbolKind.NamedType?displayProperty=nameWithType> statements. Then, put your cursor on the `AnalyzeNode` method, press **Ctrl+.** and generate the `AnalyzeNode` method.
+
+This change highlights an important design consideration for every analyzer you write: Visual Studio calls analyzers as the user is editing code. Register actions for the types of code changes that could create something your analyzer should examine. That helps minimize any performance impact for Visual Studio users that load your analyzer.
+
+Next, you should update the string resources:
```

Comment:
I would think a lot of users do not necessarily use the string resources in visual studio, give some directions. Which project are these resources in? How do I get to them?